client/dashboard/sites/settings/index.tsx

```diff
@@ -1,24 +1,33 @@
 import { useQuery } from '@tanstack/react-query';
+import { Outlet, useMatch, useMatches } from '@tanstack/react-router';
 import {
 	__experimentalHeading as Heading,
 	__experimentalVStack as VStack,
 	Card,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { siteQuery, siteSettingsQuery } from '../../app/queries';
-import { siteRoute } from '../../app/router';
+import { siteSettingsRoute } from '../../app/router';
 import PageLayout from '../../components/page-layout';
 import SubscriptionGiftingSettingsSummary from '../settings-subscription-gifting/summary';
 
 export default function SiteSettings() {
-	const { siteSlug } = siteRoute.useParams();
+	const { siteSlug } = siteSettingsRoute.useParams();
 	const { data: siteData } = useQuery( siteQuery( siteSlug ) );
 	const { data: settings } = useQuery( siteSettingsQuery( siteSlug ) );
 
+	const matches = useMatches();
+	const match = useMatch( { from: siteSettingsRoute.id } );
+
 	if ( ! siteData || ! settings ) {
 		return null;
 	}
 
+	const isExactMatch = match.id === matches[ matches.length - 1 ].id;
+	if ( ! isExactMatch ) {
+		return <Outlet />;
```

Comment:
I don't understand why <Outlet /> is needed here. It doesn't seem like the right place especially after already calling `useQuery`. Why is it needed here, but not for `${siteSlug}` (Overview) and `{$siteSlug}/sub-page`?

Reply:
I want the subpage routes (in this case `siteSettingsSubscriptionGiftingRoute`) to be a child of `siteSettingsRoute` for some reasons:

- so that they share the parent route prefix (i.e. `settings/`)
- so that they share the parent loader logic (i.e. preloading `siteSettingsQuery()`)
- (especially for this PR) so that the parent route MATCHES so its route label will be collected properly to construct the breadcrumbs. In other words, we want `/sites/${siteSlug}/settings` to be always matched in the subpages.

So, if it's an exact match, we want to render the Settings page, otherwise, we should render `<Outlet />` so that it renders the child component instead (renders the settings subpages).

Reply:
> Why is it needed here, but not for ${siteSlug} (Overview) and {$siteSlug}/sub-page?

`siteRoute` is already a parent of its children routes. It's already always shown in the subroutes. It renders as the site navigation menu (Overview, Deployments, etc).

On the other hands, we cant really compose Settings and its subpages. In `/sites/${slug}/settings` we render the list of setting summaries; in the child pages we render the settings of the subpages only. Hence the conditionals.